file.go

```diff
@@ -76,44 +75,64 @@ func (f *File) SaveAs(name string) error {
 
 // Write provides a function to write to an io.Writer.
 func (f *File) Write(w io.Writer) error {
-	_, err := f.WriteTo(w)
-	return err
-}
-
-// WriteTo implements io.WriterTo to write the file.
-func (f *File) WriteTo(w io.Writer) (int64, error) {
-	buf, err := f.WriteToBuffer()
-	if err != nil {
-		return 0, err
-	}
-	return buf.WriteTo(w)
-}
-
-// WriteToBuffer provides a function to get bytes.Buffer from the saved file.
-func (f *File) WriteToBuffer() (*bytes.Buffer, error) {
-	buf := new(bytes.Buffer)
-	zw := zip.NewWriter(buf)
-	f.calcChainWriter()
-	f.commentsWriter()
-	f.contentTypesWriter()
-	f.drawingsWriter()
-	f.vmlDrawingWriter()
-	f.workBookWriter()
-	f.workSheetWriter()
-	f.relsWriter()
-	f.styleSheetWriter()
+	var err error
+	zw := zip.NewWriter(w)
+	isZipWriterClosed := false
+	defer func() {
+		if !isZipWriterClosed {
+			zw.Close()
+		}
+	}()
 
 	for path, content := range f.XLSX {
+		if path == "xl/styles.xml" || path == "xl/workbook.xml" || path == "[Content_Types].xml" {
+			continue
+		}
 		fi, err := zw.Create(path)
 		if err != nil {
-			zw.Close()
-			return buf, err
+			return err
 		}
 		_, err = fi.Write(content)
 		if err != nil {
-			zw.Close()
-			return buf, err
+			return err
 		}
 	}
-	return buf, zw.Close()
+	err = f.calcChainWriter(zw)
```

Comment:
Since these are all functions on File, you could either:

1. Put a `writeError` field or something on File that these functions check before doing anything so that we can just call all these functions in sequence and then return `writeError` at the end.
2. Collapse these into the shorter syntax like:
```
if err := f.commentsWriter(zw); err != nil {
  return err
}
```
like you did above in some of the other functions